src/ensembl/src/content/app/custom-download/components/checkbox-with-textfields/CheckboxWithTextfields.tsx

```diff
@@ -0,0 +1,131 @@
+import React, { useState, useEffect } from 'react';
+import Checkbox from 'src/shared/checkbox/Checkbox';
+import styles from './CheckboxWithTextfields.scss';
+
+import ImageButton from 'src/shared/image-button/ImageButton';
+import { ReactComponent as AddIcon } from 'static/img/browser/zoom-in.svg';
+import PasteOrUpload from '../paste-or-upload/PasteOrUpload';
+
+export type CheckboxWithTextfieldsProps = {
+  values: string[];
+  label: string;
+  allowMultiple: boolean;
+  disabled?: boolean;
+  onChange: (values: string[]) => void;
+};
+
+const CheckboxWithTextfields = (props: CheckboxWithTextfieldsProps) => {
+  const [isChecked, setIsChecked] = useState(false);
+  const [values, setValues] = useState<(string | null)[]>(props.values);
+
+  const [shouldShowAddButton, setShowAddButton] = useState(
+    Boolean(values[values.length - 1])
+  );
```

Comment:
Is this not directly calculable from props (as done in useEffect below)? Why store it in state?

Reply:
The add button gets hidden when clicked ,displaying another input whilst the props remains the same. Does that make sense?